shell/platform/linux/public/flutter_linux/fl_event_channel.h

```diff
@@ -0,0 +1,144 @@
+ *                                     gpointer user_data) {
+ *   send_events = FALSE;
+ *   return FL_METHOD_RESPONSE (fl_method_success_response_new (NULL));
+ * }
+ *
+ * static void setup_channel () {
+ *   g_autoptr(FlStandardMethodCodec) codec = fl_standard_method_codec_new ();
+ *   channel = fl_event_channel_new (messenger, "flutter/foo",
+ *                                   FL_METHOD_CODEC (codec),
+ *                                   listen_cb, cancel_cb, NULL, NULL);
+ * }
+ * ]|
+ *
+ * #FlEventChannel matches the EventChannel class in the Flutter
+ * services library.
+ */
+
+/**
+ * FlEventChannelResponseHandle:
+ *
+ * #FlEventChannelResponseHandle is an object used to send responses
+ * with.
+ */
+
+/**
+ * FlEventChannelHandler:
+ * @channel: an #FlEventChannel.
+ * @user_data: (closure): data provided when registering this handler.
+ *
+ * Function called when the stream is listened to or cancelled.
+ *
+ * Returns: (transfer full): an #FlMethodResponse for this request.
+ */
+typedef FlMethodResponse* (*FlEventChannelHandler)(FlEventChannel* channel,
+                                                   gpointer user_data);
+
+/**
+ * fl_event_channel_new:
+ * @messenger: an #FlBinaryMessenger.
+ * @name: a channel name.
+ * @codec: the message codec.
+ * @listen_handler: (allow-none): function to call when the Dart side of the
+ * channel starts listening to the stream.
+ * @cancel_handler: (allow-none): function to call when the Dart side of the
+ * channel cancels their subscription to the stream.
+ * @user_data: (closure): user data to pass to @listen_handler and
+ * @cancel_handler.
+ * @destroy_notify: (allow-none): a function which gets called to free
+ * @user_data, or %NULL.
+ *
+ * Creates an event channel. @codec must match the codec used on the Dart
+ * end of the channel.
+ *
+ * Returns: a new #FlEventChannel.
+ */
+FlEventChannel* fl_event_channel_new(FlBinaryMessenger* messenger,
+                                     const gchar* name,
+                                     FlMethodCodec* codec,
+                                     FlEventChannelHandler listen_handler,
+                                     FlEventChannelHandler cancel_handler,
+                                     gpointer user_data,
+                                     GDestroyNotify destroy_notify);
+
+/**
+ * fl_event_channel_send:
+ * @channel: an #FlEventChannel.
+ * @event: event to send, must match what the #FlMethodCodec supports.
+ * @cancellable: (allow-none): a #GCancellable or %NULL.
+ * @error: (allow-none): #GError location to store the error occurring, or %NULL
+ * to ignore.
+ *
+ * Sends an event on the channel.
+ *
+ * Returns: %TRUE if successful.
+ */
+gboolean fl_event_channel_send(FlEventChannel* channel,
+                               FlValue* event,
+                               GCancellable* cancellable,
+                               GError** error);
+
+G_END_DECLS
+
```

Comment:
How is end of stream communicated? I didn't see a relevant constant or function.

Reply:
Added now.